src/engraving/rendering/dev/slurtielayout.cpp

```diff
@@ -1656,7 +1656,11 @@ double SlurTieLayout::noteOpticalCenterForTie(const Note* note, bool up)
         double singleSlashWidth = note->symBbox(SymId::noteheadSlashHorizontalEnds).width();
         double noteWidth = note->width();
         double center = 0.20 * note->spatium() + 0.5 * (noteWidth - singleSlashWidth);
-        return up ? note->shape().right() - center : note->shape().left() + center;
+        Shape noteShape =  note->shape();
+        noteShape.remove_if([&](ShapeElement& s) {
+            return s.item()->pos().x() + s.item()->width() > note->pos().x() + note->width();
```

Comment:
Oof, no please 😅 I think it was an error on my part to use `note->shape()` for this, because the `shape()` method includes the sub-elements (e.g. note dots), but here we should be looking exclusively at the note _head_ (so it's silly to use `shape()` to include the sub-elements and then remove them). There is a specific method `Note::headWidth` which I think should be useful here. If that doesn't work, another possibility is to get the symId and use `symWidth`. Let me know if you find a way, otherwise I'll look into it myself :)

Reply:
Good to know that ditching the shape approach should work! Thanks for reviewing, I'll try to find a better solution...

Reply:
Thanks for confirming my initial sense!  My thought was that instead of note->shape().right() it should be note->pos()->x() + note->headWidth().  And then add or subtract center as before.

I guess note->shape()->left() might not be good either and should perhaps just be note->pos()->x().  I tried to come up with a case where this might matter.  Here's 4.2.1:

![image](https://github.com/musescore/MuseScore/assets/1799936/b506924d-9801-4a2f-ae11-122ac010049c)

vs 4.1:

![image](https://github.com/musescore/MuseScore/assets/1799936/0744fa00-03ed-4e62-9a33-de5f436f01ce)

Reply:
I decided to use the `bbox` method of the notes, I think that should be the most foolproof.